engine/gamesys/src/gamesys/scripts/script_model.cpp

```diff
@@ -610,18 +610,64 @@ namespace dmGameSystem
         return 1;
     }
 
+    /*# get the AABB of the meshes
+     * Get AABB of the meshes.
+     * AABB information return as a table with `min` and `max` fields, where `min` and `max` has type `vmath.vector4`.
+     *
+     * @name model.get_aabb
+     * @param url [type:string|hash|url] the model
+     * @return aabb [type:table] A table containing info about all AABB in the format <hash(mesh_id), aabb_info>
+     * @examples
+     *
+     * ```lua
+     * model.get_aabb("#model") -> { hash("Sword") = { min = vmath.vector3(-0.5, -0.5, 0), max = vmath.vector3(0.5, 0.5, 0) }, hash("Shield") = { min = vmath.vector3(-0.5, -0.5, -0.5), max = vmath.vector3(0.5, 0.5, 0.5) } }
+     * ```
+     */
+    static int LuaModelComp_GetAabb(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        ModelComponent* component = 0;
+        dmGameObject::HInstance sender_instance = CheckGoInstance(L);
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(sender_instance);
+        dmGameObject::GetComponentFromLua(L, 1, collection, MODEL_EXT, (dmGameObject::HComponent*)&component, 0, 0);
+        if (!component)
+        {
+            return luaL_error(L, "the component '%s' could not be found", lua_tostring(L, 1));
+        }
+        lua_newtable(L);
+
+        uint32_t mesh_count = CompModelGetMeshCount(component);
+        for (uint32_t idx = 0; idx < mesh_count; ++idx)
+        {
+            dmVMath::Vector3 min, max;
+            dmhash_t mesh_id;
+            CompModelGetMeshAABB(component, idx, &mesh_id, &min, &max);
+            dmScript::PushHash(L, mesh_id);
+
+            lua_newtable(L);
+            dmScript::PushVector3(L, min);
+            lua_setfield(L, -2, "min");
+            dmScript::PushVector3(L, max);
+            lua_setfield(L, -2, "max");
+
+            lua_settable(L, -3);
+        }
+        return 1;
```

Comment:
I don't recall the design on this, but what if I wanted the AABB for the _model_, and not each individual mesh?

Reply:
Added `model.get_aabb()` for AABB of the whole model. Mesh's AABB can be retrieved via `model.get_mesh_aabb()`